lib/workos/directory_user.rb

```diff
@@ -44,6 +46,15 @@ def to_json(*)
       }
     end
 
+    def [](attribute_name)
+      # Warn the user not to use the Hash-style access for attributes.
+      warning_message = "WARNING: The Hash style access for DirectoryUser attributes is deprecated and will be removed
+in a future version. Please use user.#{attribute_name} instead of user['#{attribute_name}']"
+      puts warning_message
```

Comment:
I think we want to use [`warn`](https://ruby-doc.org/core-3.1.1/Kernel.html#method-i-warn) instead of `puts` for logging warnings.

Looks like there's also a `category` option and a `:deprecated` value that can be passed to `warn`, but I'm not sure if that option is supported on some of the older Ruby versions we apparently support.